tests/v2/validation/provisioning/rke2/etcd_bkp_restore_test.go

```diff
@@ -0,0 +1,385 @@
+	r.upgradeClusterK8sVersion(client, clusterName, k8sUpgradedVersion)
+	r.watchAndWaitForCluster(client, clusterName)
+	logrus.Infof("cluster is active again.............")
+
+	cluster, _, err := getProvisioningClusterByName(client, clusterName, r.ns)
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), k8sUpgradedVersion, cluster.Spec.KubernetesVersion)
+
+	logrus.Infof("restoring snapshot.............")
+	require.NoError(r.T(), restoreSnapshot(client, clusterName, snapshotToBeRestored, 1, "all", r.ns))
+	logrus.Infof("successfully submitted restoration request.............")
+
+	logrus.Infof("creating watch over cluster after restore.............")
+	r.watchAndWaitForCluster(client, clusterName)
+	logrus.Infof("cluster is active again.............")
+
+	logrus.Infof("creating watch over pods.............")
+	r.watchAndWaitForPods(client, clusterID)
+	logrus.Infof("All pods are up and running.............")
+
+	logrus.Infof("fetching deployment list to validate restore.............")
+	deploymentsList, err := deployments.ListDeployments(client, clusterID, &types.ListOpts{
+		Filters: map[string]interface{}{
+			"fieldSelector": "metadata.namespace=" + r.ns,
+		},
+	})
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), 1, len(deploymentsList.Items))
+	require.Equal(r.T(), w1Name, deploymentsList.Items[0].ObjectMeta.Name)
+	logrus.Infof(" deployment list validated successfully.............")
+
+	logrus.Infof("fetching ingresses list to validate restore.............")
+	ingressesList, err := ingresses.ListIngresses(client, clusterID, &types.ListOpts{
+		Filters: map[string]interface{}{
+			"fieldSelector": "metadata.namespace=" + r.ns,
+		},
+	})
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), 1, len(ingressesList.Items))
+	require.Equal(r.T(), ingress1Name, ingressesList.Items[0].ObjectMeta.Name)
+	logrus.Infof("ingress list validated successfully.............")
+
+	cluster, _, err = getProvisioningClusterByName(client, clusterName, r.ns)
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), initialK8sVersion, cluster.Spec.KubernetesVersion)
+}
+
+func (r *RKE2EtcdSnapshotRestoreTestSuite) watchAndWaitForCluster(client *rancher.Client, clusterName string) {
+	logrus.Infof("waiting for cluster to be up.............")
+	err := kwait.Poll(5*time.Second, time.Duration(defaults.WatchTimeoutSeconds)*time.Second, func() (done bool, err error) {
+		clusterObj, _, err := getProvisioningClusterByName(client, clusterName, r.ns)
+		if err != nil {
+			return false, nil
+		}
+		return clusters.IsProvisioningClusterReadyWithPoll(clusterObj)
+	})
+	require.NoError(r.T(), err)
+}
+
+func (r *RKE2EtcdSnapshotRestoreTestSuite) watchAndWaitForDeployment(client *rancher.Client, clusterID string, deploymentName string) {
+
+	logrus.Infof("waiting for deployment to be created.............")
+
+	err := kwait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		deploymentResult, err := deployments.ListDeployments(client, clusterID, &types.ListOpts{
```

Comment:
Also here why are you listing rather than doing ByID. If it's confusion you should inspect the network in browser to see how it's done. ID usually is "namespace/name". Having the list is just unnecessary.
Also the conversion would make more sense like this

```
steveclient, err = client.Steve.ProxyDownstream(clusterID)
deploymentResp, err := steveclient.SteveType(workloads.DeploymentSteveType).ByID("namespace/name")
deployment := appv1.Deployment{}
err = v1.ConvertK8s(deployment.JSONResp, deployment)
if err !=nil {
  return
]
deployment.Spec.Replicas = deployment.Status.AvailableReplicas
```